pytorch/training/docker/2.1/py3/cu121/Dockerfile.gpu

```diff
@@ -377,6 +378,9 @@ RUN /opt/conda/bin/mamba install -y python=${PYTHON_SHORT_VERSION} pytorch=${PYT
 # Install SageMaker Profiler Binary
 RUN pip install --no-cache-dir -U ${SMP_URL}
 
+# Install SM Distributed DataParallel binary
+RUN SMDATAPARALLEL_PT=1 pip install --no-cache-dir ${SMD_DATA_PARALLEL_URL}
```

Comment:
Do we have SMDataParallel tests already in the DLC repo, or is there a plan to add it in the future? I do not see specific tests being run for the new binary.

Reply:
Back in PT2.1 currency release we added this function https://github.com/aws/deep-learning-containers/blob/55bd56050fa78363d6d38e923e29bae1cff73600/test/dlc_tests/conftest.py#L908 to skip all SMDMP and SMDDP related tests. @ruhanprasad would you be able to look at all tests that are being skipped on `skip_pt21_test` fixture and remove ones that are relavant to SMDDP?

Reply:
Just made a revision to remove the `skip_pt21_test` flag on SMDDP relevant tests.  Also set the default instance type for SMDDP tests to ml.p4d.24xlarge instead of ml.p3.16xlarge as the latter is no longer supported with SMDDP